### PR TITLE
Headless support

### PR DESCRIPTION
**Pull request by ironcladlou**
I wanted to start discussion about how to extend the client/server model out to the CLI. Basically, there are a few use cases:

1. Start just the server in the foreground
2. Start just the server as a daemon
3. Connect a debugger Terminal to a server
4. Start both a server and a connected Terminal in the foreground

We should figure out what defaults and flags best facilitate all these. This PR has a small example of the "bare minimum" for IDE integrators, which is a `-headless` mode that runs just the server in the foreground and shuts down on SIGINT.

### CODE REVIEW

**Comment by derekparker** [comment]
Starting the server as a daemon is tricky, because that makes it seem we can have a server sitting around, waiting to debug something, and many clients could connect to it, be it terminal/IDE whatever. This would potentially involve some modifications to Delve to support debugging multiple processes simultaneously, or at the least supporting multiple clients connected at once.

The headless support is definitely something we want asap, which would be essential for IDE/Editor integration.

File: cmd/dlv/main.go [hunk 1]
```diff
@@ -120,15 +124,28 @@ func main() {
...
+	if !headless {
+		// Create and start a terminal
+		client := rest.NewClient(listener.Addr().String())
+		term := terminal.New(client)
+		err, status = term.Run()
+	} else {
+		ch := make(chan os.Signal)
+		signal.Notify(ch, sys.SIGINT)
+	runLoop:
+		for {
```
Comment (derekparker) [inline cmd/dlv/main.go]: Since we break immediately once we get a signal, is there need for the loop? Just block on the signal and stop the server.

Comment (ironcladlou) [inline cmd/dlv/main.go]: I must have been very tired...

**Comment by ironcladlou** [comment]
I think this is the minimum viable thing for integrations. A reasonable followup PR would be another flag which starts a terminal connected to a specified headless server.

>Starting the server as a daemon is tricky, because that makes it seem we can have a server sitting around, waiting to debug something, and many clients could connect to it, be it terminal/IDE whatever. This would potentially involve some modifications to Delve to support debugging multiple processes simultaneously, or at the least supporting multiple clients connected at once.

I definitely don't want to imply or support multiple client support, although it's technically safe-ish at the moment due to our protection of the DebuggedProcess using the operations channel. Daemon mode probably doesn't make much sense.

**Comment by derekparker** [comment]
Merged.

**Pull request closed without merging**